app/controllers/static_controller.rb

```diff
@@ -27,9 +27,40 @@ class StaticController < ApplicationController
   }
   CUSTOM_PAGES = {} # Add via `#add_topic_static_page` in plugin API
 
+  def validate_redirect_param
```

Comment:
When we start a method with `validate_` it usually means it will check for "something" and then either do nothing or raise an error if something's wrong.

In this case, we want to return a `path` extracted from the `redirect` query parameter, so how about `extract_redirect_path`?